libraries/rush-lib/src/cli/RushXCommandLine.ts

```diff
@@ -45,129 +47,139 @@ interface IRushXCommandLineArguments {
-      if (args.commandArgs.length > 0) {
-        // This approach is based on what NPM 7 now does:
-        // https://github.com/npm/run-script/blob/47a4d539fb07220e7215cc0e482683b76407ef9b/lib/run-script-pkg.js#L34
-        const escapedRemainingArgs: string[] = args.commandArgs.map((x) => Utilities.escapeShellParameter(x));
+      console.log(`Use ${colors.yellow('"rushx --help"')} for more information.`);
+      return;
+    }
 
-        commandWithArgs += ' ' + escapedRemainingArgs.join(' ');
+    let commandWithArgs: string = scriptBody;
+    let commandWithArgsForDisplay: string = scriptBody;
+    if (args.commandArgs.length > 0) {
+      // This approach is based on what NPM 7 now does:
+      // https://github.com/npm/run-script/blob/47a4d539fb07220e7215cc0e482683b76407ef9b/lib/run-script-pkg.js#L34
+      const escapedRemainingArgs: string[] = args.commandArgs.map((x) => Utilities.escapeShellParameter(x));
 
-        // Display it nicely without the extra quotes
-        commandWithArgsForDisplay += ' ' + args.commandArgs.join(' ');
-      }
+      commandWithArgs += ' ' + escapedRemainingArgs.join(' ');
 
-      if (!args.quiet) {
-        console.log(`> ${JSON.stringify(commandWithArgsForDisplay)}\n`);
-      }
+      // Display it nicely without the extra quotes
+      commandWithArgsForDisplay += ' ' + args.commandArgs.join(' ');
+    }
 
-      const packageFolder: string = path.dirname(packageJsonFilePath);
-
-      const exitCode: number = Utilities.executeLifecycleCommand(commandWithArgs, {
-        rushConfiguration,
-        workingDirectory: packageFolder,
-        // If there is a rush.json then use its .npmrc from the temp folder.
-        // Otherwise look for npmrc in the project folder.
-        initCwd: rushConfiguration ? rushConfiguration.commonTempFolder : packageFolder,
-        handleOutput: false,
-        environmentPathOptions: {
-          includeProjectBin: true
-        }
-      });
+    if (!args.quiet) {
+      console.log(`> ${JSON.stringify(commandWithArgsForDisplay)}\n`);
+    }
 
-      if (exitCode > 0) {
-        console.log(colors.red(`The script failed with exit code ${exitCode}`));
+    const packageFolder: string = path.dirname(packageJsonFilePath);
+
+    const exitCode: number = Utilities.executeLifecycleCommand(commandWithArgs, {
+      rushConfiguration,
+      workingDirectory: packageFolder,
+      // If there is a rush.json then use its .npmrc from the temp folder.
+      // Otherwise look for npmrc in the project folder.
+      initCwd: rushConfiguration ? rushConfiguration.commonTempFolder : packageFolder,
+      handleOutput: false,
+      environmentPathOptions: {
+        includeProjectBin: true
       }
+    });
 
-      process.exitCode = exitCode;
-    } catch (error) {
-      console.log(colors.red('Error: ' + (error as Error).message));
+    if (exitCode > 0) {
+      console.log(colors.red(`The script failed with exit code ${exitCode}`));
     }
```

Comment:
Nearly the entire contents of this internal method were wrapped in the try/catch, so I moved that up to the caller.  Again, this makes the internal version easier to test.   

I think the process.exitCode should probably move up to the caller as well.

Reply:
If so, have `exitCode` be the return value.

Reply:
OK, what I did instead is replace the returns with throws (where there is an error).

Reply:
@octogonz , @dmichon-msft Please mark this thread resolved if you are happy with the decision to throw instead of returning exit codes.

Reply:
As long as you tested it and it works, I'm fine with it.